### PR TITLE
Run command

### PR DESCRIPTION
**Pull request by tasuki**
Completes #47.

### CODE REVIEW

**Comment by sroze** [comment]
It's not really fixing #47 as the proposed command interface is not the same :)

Actually, IMO the container name should be guessed when possible and, if not, prompted to the user (and or by letting him specify directly the container by a `-c` option for instance).

**Comment by tasuki** [comment]
@sroze You're right, the interface is not the same.

How to guess the container name? When we have more than one container, this appears impossible from my point of view.

I strongly dislike the `-c` option, even though had it for a while already. The `docker-compose` way of having the service as a mandatory parameter just seemed to make more sense.

Also, there's the issue of the executed command having to be one argument (ie wrapped in quotes when multiword). I'm not sure how to get past that in Symfony, but will investigate further!

**Comment by sroze** [comment]
@tasuk we can guess the container name based on the `build` parameter of the `docker-compose.yml` file. The goal of this `dock-cli` is to be more clever than the `docker-compose` command line tool, in case of in-project usages IMO, so we should be able to do that:
- If the user `pwd` is _in_ the `build` directory of a container, just use it.
- Elseif the user gave the name of the container, use it.
- Else, ask the name of the container ([based on a list](http://symfony.com/doc/current/components/console/helpers/questionhelper.html#let-the-user-choose-from-a-list-of-answers))

Moreover, a mandatory service parameter is a lot of repetition in the commands the developer have to run in the app development life-cycle, which is one of the _overhead_ introduced by Docker compared to VMs used right now at @Inviqa.

To be able to have commands as many arguments, you can use the [`InputArgument::IS_ARRAY` input type](http://symfony.com/doc/current/components/console/introduction.html#using-command-arguments) and then `implode` them to have the command to run.

**Comment by tasuki** [comment]
Awesome, thanks for all the info!

**Comment by richardmiller-zz** [comment]
If the commands are likely to be run on the container identified by the build parameter then it would be good to just use that without having to specify it and then specify when wanting to run it on a different container.

**Comment by tasuki** [comment]
@sroze: I did the logic a little differently, as it seemed more logical:
- If the user gave the name of the container, use it.
- Elseif the user pwd is in the build directory of a container, just use it.
- Else, ask for the name of the container.

File: src/Cli/RunCommand.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        if ($input->getOption('service') !== null) {
+            $service = $input->getOption('service');
+        } else {
+            try {
+                $service = $this->config->getCurrentService();
+            } catch (\Exception $e) {
```
Comment (sroze) [inline src/Cli/RunCommand.php]: Could you use a custom exception here instead of directly `\Exception` ?

File: src/Cli/RunCommand.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+     */
+    private $processRunner;
+
+    /**
+     * @var Config
+     */
+    private $config;
+
+    /**
+     * @param ProcessRunner $processRunner
```
Comment (sroze) [inline src/Cli/RunCommand.php:28]: Missing the `$config` phpdoc

File: composer.json [hunk 3]
```diff
@@ -9,6 +9,8 @@
     "pimple/pimple": "^3.0",
     "symfony/event-dispatcher": "~2.7",
     "behat/behat": "^3.0",
+    "phpspec/phpspec": "~2.0",
```
Comment (sroze) [inline composer.json]: Can you move the dependency to `dev` ?

**Comment by sroze** [comment]
Really good @tasuk ! :+1:

File: composer.json (hunk 3 above)
Comment (tasuki) [inline composer.json]: Wanted to do that, but then saw behat in non-dev and decided to put phpspec alongside it. Move behat to `dev` as well?

Comment (sroze) [inline composer.json]: Sure, it's a mistake to have behat in normal dependencies :)

**Comment by sroze** [comment]
@tasuk just missing the documentation file (based on the other template, no big needs now) and LGTM !

**Comment by sroze** [comment]
:+1:

**Pull request merged**